liquibase-core/src/test/groovy/liquibase/changelog/visitor/ValidatingVisitorTest.groovy

```diff
@@ -0,0 +1,272 @@
+        changeSet1.addChange(new CreateTableChange() {
+            @Override
+            void finishInitialization() throws SetupException {
+                throw new SetupException("Test message")
+            }
+        });
+
+        ValidatingVisitor handler = new ValidatingVisitor(new ArrayList<RanChangeSet>())
+        handler.visit(changeSet1, new DatabaseChangeLog(), null, null)
+
+        then:
+        handler.getSetupExceptions().size() == 1
+        handler.getSetupExceptions().get(0).getMessage().equalsIgnoreCase("Test message")
+        !handler.validationPassed()
+    }
+
+    void "validate visit error"() throws Exception {
+        when:
+        def changeSet1 = new ChangeSet("1", "testAuthor", false, false, "path/changelog", null, null, null)
+        changeSet1.addChange(new CreateTableChange() {
+            @Override
+            ValidationErrors validate(Database database) {
+                ValidationErrors changeValidationErrors = new ValidationErrors()
+                changeValidationErrors.addError("Test message")
+                return changeValidationErrors;
+            }
+        })
+
+        List<RanChangeSet> ran = new ArrayList<RanChangeSet>()
+        ValidatingVisitor handler = new ValidatingVisitor(ran)
+        handler.visit(changeSet1, new DatabaseChangeLog(), null, null)
+
+        then:
+        handler.getValidationErrors().getErrorMessages().size() == 1
+        handler.getValidationErrors().getErrorMessages().get(0).startsWith("Test message")
+        !handler.validationPassed()
+    }
+
+    void "validate visit duplicate"() throws Exception {
+        when:
+        def changeSet1 = new ChangeSet("1", "testAuthor", false, false, "path/changelog", null, null, null)
+        ValidatingVisitor handler = new ValidatingVisitor(new ArrayList<RanChangeSet>())
+        handler.visit(changeSet1, new DatabaseChangeLog(), null, null)
+        handler.visit(changeSet1, new DatabaseChangeLog(), null, null)
+
+        then:
+        handler.getDuplicateChangeSets().size() == 1
+        !handler.validationPassed()
+    }
+
+    void "validate error on empty author when strict configuration is set as true"() {
+        when:
+        ChangeSet changeSet = new ChangeSet("emptyAuthor", "", false, false, "path/changelog", null, null, null)
+        ValidatingVisitor handler = new ValidatingVisitor(new ArrayList<>())
+
+        Scope.child([
+                (GlobalConfiguration.STRICT.getKey()): Boolean.TRUE,
+        ], new Scope.ScopedRunner() {
+            @Override
+            void run() throws Exception {
+                handler.visit(changeSet, new DatabaseChangeLog(), null, null)
+            }
+        })
+
+        then:
+         handler.getValidationErrors().getErrorMessages().get(0).contains("ChangeSet Author is empty")
+    }
+
+    void "validate error on empty author when strict configuration is set as false"() {
```

Comment:
A more accurate test description would be "validate *there is no* error on empty author when strict configuration is set as false".

Reply:
Sounds good to me, I'll make this update in a bit.